docs/usage/inference.rst

```diff
@@ -0,0 +1,168 @@
+Inference
+=========
+
+The inference functionality in mlGeNN allows you to run trained networks efficiently.
+This section covers how to use the inference compiler and its compiled networks.
+
+Creating an Inference Compiler
+----------------------------
+The ``InferenceCompiler`` provides the core functionality for running inference on
+trained networks. Here's a basic example:
+
+.. code-block:: python
+
+    from ml_genn.compilers import InferenceCompiler
+
+    # Create compiler
+    compiler = InferenceCompiler(example_timesteps=500)
+
+    # Compile network
+    compiled_net = compiler.compile(network, inputs=net_inputs, outputs=net_outputs)
+
+Key Methods
+----------
+
+predict()
+~~~~~~~~
+The ``predict()`` method runs inference on input data and returns predictions:
```

Comment:
can you use:

```python
.. automethod:: ml_genn.compilers.InferenceCompiler.predict
    :noindex:
```

to pull through the existing documentation from the source for these methods? Otherwise it becomes something of a maintenance nightmare!